### ISSUE TITLE
Bootstrap: OpenShift support and documentation

### ISSUE THREAD

**Issue by longmuir** [akka/akka-management#175]
Provide distinct documentation and support for Openshift deployments for Service Discovery and Cluster formation. This may be already mostly handled with vanilla Kubernetes, but it's worth having distinct instructions. This needs to be confirmed.

**Comment by ktoso** [comment]
Agreed that it would be very good to get to this eventually

**Comment by ktoso** [comment]
@longmuir seems to confirm it worked well on an openshift deployment -- assuming it was the k8s API based discovery used?

**Comment by longmuir** [comment]
Yes, we confirmed that there were no issues with using the K8s API discovery method on an OpenShift cluster. I will try to confirm which version of OpenShift they were using. We did not try the headless service mechanism, as the K8s API method seemed favourable for allowing cluster formation to occur before pods declare themselves ready and can be included in a Service.

For the headless service mechanism, it would be worth confirming that it works using something like OpenShift Origins - https://docs.openshift.org/ to validate it. If it works here as well, I think we are safe to document that the K8s API and Headless service mechanisms also apply for OpenShift.

It's worth noting that this customer did not use Lightbend Orchestration(they use Maven which wasn't supported at the time), and thus wrote their own K8s YAML definitions according to the example in the Akka Discovery docs. Thus, this is only confirming that Akka Discovery, Bootstrap and the Akka Management HTTP worked with OpenShift. As Lightbend Orchestration makes much more extensive use of the K8s APIs, it will require more extensive testing to confirm that it supports Openshift.
